src/PowerShell/Microsoft.WinGet.DSC/Microsoft.WinGet.DSC.psm1

```diff
@@ -193,130 +187,169 @@ class WinGetAdminSettings
 }
 
 [DSCResource()]
-class WinGetSources
+class WinGetSource
 {
-    # We need a key. Do not set.
-    [DscProperty(Key)]
-    [string]$SID
-
-    # An array of Hashtable with the key value properties that follows the source's group policy schema.
+    [DscProperty(Key, Mandatory)]
+    [ValidateNotNullOrWhiteSpace()]
+    [string]$Name
+    
     [DscProperty(Mandatory)]
-    [Hashtable[]]$Sources
-
+    [string]$Argument

```

Comment:
do we need `[ValidateNotNullOrWhiteSpace()]` here?

Reply:
No, because if `Ensure = Absent` we don't want to require an argument value (although from my testing it seems like the validation only happens during assignment).  If someone provides an empty argument they will still get an error, just from deeper in the stack.